### PR TITLE
Php Mail 

### PR DESCRIPTION
**Pull request by sowmiyasubramaniyan**
Php mail  with from name option
need to check all the mail-sending functions
 **Note**:
Run the following command: php artisan migrate. 
For PHP mail configuration, you need to configure the following mail ID in "cp-7" on the "qafaveo" server: [test@qa.faveodemo.com](mailto:test@qa.faveodemo.com).

### CODE REVIEW

**Comment by bhanu2217** [comment]
@sowmiyasubramaniyan Please connect with me and show me demo of this page and take my feedback

File: app/Console/Kernel.php [hunk 1]
```diff
@@ -168,28 +167,14 @@ protected function commands()
     public function cloudEmail()
     {
         try {
-            $settings = Setting::find(1);
+            $setting = Setting::find(1);
             $mail = new \App\Http\Controllers\Common\PhpMailController();
-            $mailer = $mail->setMailConfig($settings);
             $clouds = cloudemailsend::all();
```
Comment (RafficMohammed) [inline app/Console/Kernel.php:172]: use cursor() instead of all()

File: app/Http/Controllers/Auth/BaseAuthController.php [hunk 2]
```diff
@@ -179,19 +178,31 @@ public function sendActivation($email, $method, $str = '')
             }
 
             $url = url("activate/$token");
-
+            //check in the settings
+            $settings = new \App\Model\Common\Setting();
+            $settings = $settings->where('id', 1)->first();
+
```
Comment (RafficMohammed) [inline app/Http/Controllers/Auth/BaseAuthController.php:184]: 🚩

File: app/Http/Controllers/Auth/ForgotPasswordController.php [hunk 3]
```diff
@@ -45,61 +44,59 @@ public function showLinkRequestForm()
...
             } else {
-                $activate = $password->create(['email' => $email, 'token' => $token, 'created_at' => \Carbon\Carbon::now()]);
+                $activate = $password->create(['email' => $email, 'token' => $token, 'created_at'=>\Carbon\Carbon::now()]);
                 $token = $activate->token;
             }
 
             $url = url("password/reset/$token");
 
-            $to = $user->email;
+            $user = new \App\User();
```
Comment (RafficMohammed) [inline app/Http/Controllers/Auth/ForgotPasswordController.php:62]: 🚩

**Review by RafficMohammed — CHANGES_REQUESTED** [review]
Discuss with me

**Comment by sandesh556** [comment]
change the label after emails are handled through jobs

**Comment by sandesh556** [comment]
rebase and convert it for cloud also

**Comment by sowmiyasubramaniyan** [comment]
> rebase and convert it for cloud also

Done

File: app/Http/Controllers/Tenancy/TenantController.php [hunk 4]
```diff
@@ -192,7 +192,7 @@ public function createTenant(Request $request)
                 'domain.regex' => 'Special characters are not allowed in domain name',
             ]);
 
-        $settings = Setting::find(1);
+        $setting = Setting::find(1);
```
Comment (sandesh556) [inline app/Http/Controllers/Tenancy/TenantController.php:194]: don't change the variable name

File: app/Http/Controllers/Tenancy/TenantController.php [hunk 5]
```diff
@@ -283,13 +283,8 @@ public function createTenant(Request $request)
                     $result->message = str_replace('website', strtolower($product), $result->message);
                     $userData = $result->message.'<br><br> Email:'.' '.$user.'<br>'.'Password:'.' '.$result->password;
                     $this->prepareMessages($faveoCloud, $user, true);
-                    $email = (new Email())
-                        ->from($settings->email)
-                        ->to($user)
-                        ->subject($template->name)
-                        ->html($mail->mailTemplate($template->data, $templatevariables = ['message' => $userData, 'name' => \Auth::user()->first_name.' '.\Auth::user()->last_name]));
+                    $mail->SendEmail($setting->email, $user, $userData, 'New instance created');
```
Comment (sandesh556) [inline app/Http/Controllers/Tenancy/TenantController.php:284]: Also how will the template be sent?

**Review by sandesh556 — CHANGES_REQUESTED** [review]

**Comment by sandesh556** [comment]
i see that variable name is changed in few places

**Comment by bhanu2217** [comment]
Looks good. Lets proceed.

**Review by sandesh556 — APPROVED** [review]

**Review by RafficMohammed — DISMISSED** [review]

**Review by SarfrazS — CHANGES_REQUESTED** [review]
I Configured php mail and then server team has configured horizon. After that i purchase products. Mail  receiving but pending job, complete job and  failed job are not keep on loading....
image

**Review by SarfrazS — CHANGES_REQUESTED** [review]
- [ ] Issue1:- With redis Registration conformation and verification mail are not sending.
- [ ] Issue2:- With redis invoice generation mail not sending.
- [ ] Issue3:- With redis Remider and expiry mail not sending.
- [ ] Issue4:-  Horizon not working.
Query:- Driver i selected php mail now i entered gmail id its accepting is it correct.
image

**Comment by sowmiyasubramaniyan** [comment]
@SarfrazS mails are going with Redis please check again. Except horizon mentioned issues are fixed.

**Review by SarfrazS — APPROVED** [review]
Perpetual products notification checked, all are sending with redis queue from PHP mail driver.
image
image
image

Note:- Cloud notification and horizon will check on a separate instance.

**Comment by SarfrazS** [comment]
@sowmiyasubramaniyan Please resolve conflicts and let us know.

**Comment by SarfrazS** [comment]
@sowmiyasubramaniyan Please resolve the conflicts and let me know.
image

**Comment by Sabaric12** [comment]
While Making the Payment we are getting this below error
Screenshot from 2023-08-25 12-00-52

**Comment by Sabaric12** [comment]
Tested this pr with the Server instance 
Now all the mentioned issues are working fine now
Screenshot from 2023-08-25 18-28-06
Screenshot from 2023-08-25 18-28-24

**Review by Sabaric12 — DISMISSED** [review]

**Comment by Dhayanan** [comment]
Tested this PR on existing DB on server:-
Mentioned issues are working fine. now  we are able recevie the mail for all the template
image

image

**Review by Dhayanan — DISMISSED** [review]

**Comment by sandesh556** [comment]
@sowmiyasubramaniyan get a free cloud environment from thiru then we'll do this

**Comment by Sabaric12** [comment]
Tested this pr with the Cloud 
Screenshot from 2023-08-29 20-01-57
Screenshot from 2023-08-29 20-04-22
Server instance
Now all the mentioned issues are working fine now
Note:
The Horizon alone not fixed developer told it will be fixed on next release and in the mail template changes the name that also developer said she will fix in next release

**Review by Sabaric12 — APPROVED** [review]

**Comment by SarfrazS** [comment]
@Sabaric12 Have you tested with php mail driver?

**Comment by SarfrazS** [comment]
@sowmiyasubramaniyan Except Logo all issues are fixed. Once the logo issue fixes please let me know will retest it.

**Review by SarfrazS — DISMISSED** [review]
Now company logo showing on mail  
 
image

**Pull request merged**